src/v/io/persistence.cc

```diff
@@ -104,17 +165,23 @@ disk_persistence::allocate(uint64_t alignment, size_t size) noexcept {
 
 seastar::future<seastar::shared_ptr<persistence::file>>
 disk_persistence::create(std::filesystem::path path) noexcept {
+    return maybe_fail_create().then([path = std::move(path)] {
     using of = seastar::open_flags;
     const auto flags = of::create | of::exclusive | of::rw;
-    auto file = co_await seastar::open_file_dma(path.string(), flags);
-    co_return seastar::make_shared<disk_file>(std::move(file));
+    return seastar::open_file_dma(path.string(), flags).then([](auto file) {
+    return seastar::make_ready_future<seastar::shared_ptr<persistence::file>>(seastar::make_shared<disk_file>(std::move(file)));
+    });
+    });
 }
 
 seastar::future<seastar::shared_ptr<persistence::file>>
 disk_persistence::open(std::filesystem::path path) noexcept {
+    return maybe_fail_open().then([path = std::move(path)] {
     const auto flags = seastar::open_flags::rw;
-    auto file = co_await seastar::open_file_dma(path.string(), flags);
-    co_return seastar::make_shared<disk_file>(std::move(file));
+    return seastar::open_file_dma(path.string(), flags).then([](auto file) {
+    return seastar::make_ready_future<seastar::shared_ptr<persistence::file>>(seastar::make_shared<disk_file>(std::move(file)));
+    });
+    });
```

Comment:
What happened to coro4lyfe?

Reply:
haha. given this is the bottom of the stack, every I/O will incur the performance cost of coroutine (see Travis' talk a couple months ago).